LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayloadJoinRequest.cs

```diff
@@ -121,7 +117,9 @@ public override byte[] GetByteMessage()
             messageArray.AddRange(Mhdr.ToArray());
             messageArray.AddRange(AppEUI.ToArray());
             messageArray.AddRange(DevEUI.ToArray());
-            messageArray.AddRange(DevNonce.ToArray());
+            var buffer = new byte[DevNonce.Size];
```

Comment:
If we can determine the output buffer length in advance, then why not copy to the array instead of a dynamic list, array conversions and then here an intermediate two-byte array allocation for the nonce?

Reply:
Done in fd28236af32ae579bb5c855c957271aba387e04b